### PR TITLE
JWT with RS256

### LINKED ISSUE #8: Uses RS256 instead of HS256 for JWT signature algorithm
## Today

Now I use HS256 where it uses a single shared key to sign and verify JWT token signature. it is simple but requires configuration option on both `travis-ci/travis-build` and this feature.

## Proposal

Uses RS256 where it uses an RSA private key (available in the context of a Travis build) to sign JWT token, artifacts server will use RSA public key (available when querying Travis Postgres database) to verify JWT token signatures.

The implementation will be more work but no configuration required on neither `travis-build` nor artifacts.

/cc @meatballhat

### PR DESCRIPTION
**Pull request by shawnzhu**
fixes #8 

It introduces one configuration option to specify a public key file path via environment variable `JWT_PUBLIC_KEY_PATH`

### CODE REVIEW

File: router/jwt.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/dgrijalva/jwt-go/request"
+	"github.com/gin-gonic/gin"
+)
+
+// Auth authenticates request validating JWT signature by a given public key.
+// a user agent must provide a BEARER token via the Authorization header
+func Auth() gin.HandlerFunc {
+	publicKeyPEM, _ := ioutil.ReadFile(os.Getenv("JWT_PUBLIC_KEY_PATH"))
```
Comment (meatballhat) [inline router/jwt.go]: I would prefer these to be strings read directly from the env.  We typically deploy via Heroku and Docker, injecting configuration entirely via the env, and making no assumptions about file access.

File: router/route.go [hunk 2]
```diff
@@ -13,19 +11,16 @@ import (
 
 // Routes load middlewares
 func Routes(middleware ...gin.HandlerFunc) http.Handler {
-	var (
-		router    = gin.Default()
-		jwtSecret = os.Getenv("JWT_SECRET")
-	)
+	var router = gin.Default()
```
Comment (meatballhat) [inline router/route.go:14]: I realize this may be a little late, but are you OK with using gorilla mux for routing instead?

Comment (meatballhat) [reply router/route.go:14]: I also prefer negroni for middleware handling 😐 .  Sorry to be so specific.   If you feel strongly about sticking with gin, then I'll roll with it.

**Review by meatballhat — CHANGES_REQUESTED** [review]

File: router/jwt.go (hunk 1 above)
Comment (shawnzhu) [reply router/jwt.go]: Yes! that's what I'm going to do next since I will have the same problem when deploying it.

File: router/route.go (hunk 2 above)
Comment (shawnzhu) [reply router/route.go:14]: No problem. but I will make it in another PR

**Review by meatballhat — APPROVED** [review]
👏 🎉

**Pull request merged**